site_scons/prereq_tools/base.py

```diff
@@ -274,27 +274,20 @@ def run_commands(self, commands, subdir=None):
         if not self.env:
             raise Exception("PreReqComponent not initialized")
         retval = True
-        old = os.getcwd()
-        if subdir:
-            if self.__dry_run:
-                print('Would change dir to %s' % subdir)
-            else:
-                os.chdir(subdir)
 
-        print('Running commands in %s' % os.getcwd())
+        if subdir:
+            print('Running commands in {}'.format(subdir))
         for command in commands:
             command = self.env.subst(command)
             if self.__dry_run:
                 print('Would RUN: %s' % command)
                 retval = True
             else:
                 print('RUN: %s' % command)
-                if subprocess.call(command, shell=True,   # nosec
-                                   env=self.env['ENV']) != 0:
+                cmd = command.split(' ')
```

Comment:
I could go further with this and allow commands to be passed in as arrays anyway which would be nicer, but I stumbled upon this improvement accidentally and want to get back to development work.

Reply:
Done, all commands should now correctly be defined as arrays, and ' ' isn't used as a command delimiter anywhere.  I've not checking installing into paths with spaces in and I'd probably still expect issues somewhere but this code should be correct in all cases now.